romancal/dq_init/dqflags.py

```diff
@@ -0,0 +1,116 @@
+         'NO_FLAT_FIELD':    2**18,  # Flat field cannot be measured
+         'NO_GAIN_VALUE':    2**19,  # Gain cannot be measured
+         'NO_LIN_CORR':      2**20,  # Linearity correction not available
+         'NO_SAT_CHECK':     2**21,  # Saturation check not available
+         'UNRELIABLE_BIAS':  2**22,  # Bias variance large
+         'UNRELIABLE_DARK':  2**23,  # Dark variance large
+         'UNRELIABLE_SLOPE': 2**24,  # Slope variance large (i.e., noisy pixel)
+         'UNRELIABLE_FLAT':  2**25,  # Flat variance large
+         'RESERVED_5':       2**26,  #
+         'RESERVED_6':       2**27,  #
+         'UNRELIABLE_RESET': 2**28,  # Sensitive to reset anomaly
+         'RESERVED_7':       2**29,  #
+         'OTHER_BAD_PIXEL':  2**30,  # A catch-all flag
+         'REFERENCE_PIXEL':  2**31,  # Pixel is a reference pixel
+         }
+
+# Group-specific flags. Once groups are combined, these flags
+# are equivalent to the pixel-specific flags.
+group = {'GOOD':       pixel['GOOD'],
+         'DO_NOT_USE': pixel['DO_NOT_USE'],
+         'SATURATED':  pixel['SATURATED'],
+         'JUMP_DET':   pixel['JUMP_DET'],
+         'DROPOUT':    pixel['DROPOUT'],
+         }
+
+def interpret_bit_flags(bit_flags, flip_bits=None, mnemonic_map=pixel):
+    """Converts input bit flags to a single integer value (bit mask) or `None`.
+
+    Wraps `astropy.nddate.bitmask.interpret_bit_flags`, allowing the
+    bit mnemonics to be used in place of integers.
+
+    Parameters
+    ----------
+    bit_flags : int, str, list, None
+        See `astropy.nddate.bitmask.interpret_bit_flags`.
+        Also allows strings using Roman mnemonics
+
+    flip_bits : bool, None
+        See `astropy.nddata.bitmask.interpret_bit_flags`.
+
+    mnemonic_map : {str: int[,...]}
+        Dictionary associating the mnemonic string to an integer value
+        representing the set bit for that mnemonic.
+
+    Returns
+    -------
+    bitmask : int or None
+        Returns an integer bit mask formed from the input bit value or `None`
+        if input ``bit_flags`` parameter is `None` or an empty string.
+        If input string value was prepended with '~' (or ``flip_bits`` was set
+        to `True`), then returned value will have its bits flipped
+        (inverse mask).
+    """
+    return dqflags.interpret_bit_flags(bit_flags, flip_bits, mnemonic_map)
+
+def dqflags_to_mnemonics(dqflags, mnemonic_map=pixel):
+    """Interpret value as bit flags and return the mnemonics
+
+    Parameters
+    ----------
+    dqflags : int-like
+        The value to interpret as DQ flags
+
+    mnemonic_map: {str: int[,...]}
+        Dictionary associating the mnemonic string to an integer value
+        representing the set bit for that mnemonic.
+
+    Returns
+    -------
+    mnemonics : {str[,...]}
+        Set of mnemonics represented by the set bit flags
+
+    """
+    return dqflags.dqflags_to_mnemonics(dqflags, mnemonic_map)
```

Comment:
Why not just call the methods in stcal instead of this roundabout way and keep only `pixel` and `group` in this file?

Reply:
I thought I had already done that, but I fixed it now. Moved to imports from STCal.